File: arrays/brick-wall/brick-wall.go
package brickwall

func leastBricks(wall [][]int) int {
	count := map[int]int{}

	for _, list := range wall {
		edge := 0
		for i := 0; i < len(list)-1; i++ {
			edge += list[i]
			if _, ok := count[edge]; !ok {
				count[edge] = 0
			}
			count[edge]++
		}
	}

	max := 0

	for _, v := range count {
		if v > max {
			max = v
		}
	}
	return len(wall) - max
}

File: arrays/brick-wall/brick-wall.js
/**
 * @param {number[][]} wall
 * @return {number}
 */
function leastBricks(wall) {
  const count = {};

  for (let list of wall) {
    let edge = 0;
    for (let i = 0; i < list.length - 1; i++) {
      edge += list[i];

      if (count[edge] === undefined) {
        count[edge] = 0;
      }

      count[edge]++;
    }
  }

  let max = 0;
  for (let key in count) {
    max = Math.max(max, count[key]);
  }

  return wall.length - max;
}
